### PR TITLE
A better way too solve the problem of special characters

### PR DESCRIPTION
**Pull request by Florent2305**
A better way too solve the problem of special characters
+ some syntax improvements like indentation and "for :"

Need a check on linux or OS X

### CODE REVIEW

File: src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp [hunk 1]
```diff
@@ -192,9 +192,9 @@ void medAbstractDatabaseImporter::importFile ( void )
     bool atLeastOneImportSucceeded = false;
     bool atLeastOneImportError = false;
 
-    foreach ( QString file, fileList )
+    for( QString file: fileList )
     {
-        if ( d->isCancelled ) // check if user cancelled the process
+        if ( d->isCancelled ) // check if user canceled the process
```
Comment (mathildemerle) [inline src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp:197]: both seams to be ok: https://grammarist.com/spelling/cancel/

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp [hunk 2]
```diff
@@ -165,30 +164,37 @@ void DCMTKImageIO::ReadImageInformation()
...
-        ++fileIndex;
+            ++fileIndex;
+        }
     }
 
+
     fileIndex = 0;
 
-    for (NameSetType::const_iterator it = fileNamesSet.begin(); it!= fileNamesSet.end(); it++)
+    //for (NameSetType::const_iterator it = fileNamesSet.begin(); it!= fileNamesSet.end(); it++)
```
Comment (mathildemerle) [inline src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: to remove

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp [hunk 3]
```diff
@@ -218,13 +224,13 @@ void DCMTKImageIO::ReadImageInformation()
        In the next loop, slices are ordered according to their instance number, in case multiple
        volumes are found.
      */
-    for (SliceLocationSetType::const_iterator l = m_LocationSet.begin(); l!=m_LocationSet.end(); l++)
+    //for (SliceLocationSetType::const_iterator l = m_LocationSet.begin(); l != m_LocationSet.end(); l++)
```
Comment (mathildemerle) [inline src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: to remove

File: src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp (hunk 1 above)
Comment (Florent2305) [reply src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp:197]: ok it's just my orthographic corrector to suggest this

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp (hunk 3 above)
Comment (Florent2305) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: Done

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp (hunk 2 above)
Comment (Florent2305) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: Done

**Review by mathildemerle — APPROVED** [review]

**Review by mjuhoor — APPROVED** [review]

**Pull request merged**